src/Tenancy/Identification/TenantResolver.php

```diff
@@ -179,4 +183,60 @@ protected function retrieveDriverMethods(string $driver): array
     {
         return (new ReflectionClass($driver))->getMethods(ReflectionMethod::IS_PUBLIC);
     }
+
+    protected function identifyByContract(string $contract)
+    {
+        /** @var Tenant|null $tenant */
+        $tenant = $this->events()->until(new Events\Resolving($models = $this->getModels()));
+
+        if (!$tenant && count($this->drivers) > 0) {
+            $tenant = $this->resolveFromDriver($models, $contract);
+        }
+
+        if ($tenant) {
+            $this->events()->dispatch(new Events\Identified($tenant));
+        }
+
+        // Provide a debug log entry when no tenant was identified, possibly because no identification driver is active.
+        if (!$tenant && count($this->drivers) === 0) {
+            logger('No tenant was identified, a possible cause being that no identification drivers are available.');
+        }
+
+        if (!$tenant) {
+            $this->events()->dispatch(new Events\NothingIdentified($tenant));
+        }
+
+        $this->events()->dispatch(new Events\Resolved($tenant));
+
+        return $tenant;
+    }
```

Comment:
This needs to be looked at, it's almost a complete copy paste of the invoke method